### ISSUE TITLE
Office.OnReady is getting called twice

### ISSUE THREAD

**Issue by kakusharma** [OfficeDev/office-js#659]
The Office.OnReady function is called twice for an Outlook Contexxtual Add-In that I am creating. 

## Expected Behavior

 Office.OnReady should be called only once.

## Current Behavior

Office.OnReady should be called only once

## Steps to Reproduce, or Live Example

* Link to live example: ______

* Additional details:
  1.  Office.onReady().then(info => {  }); This is getting called twice. No special steps to reproduce. Its just like plugging the OnReady in the add-in and its behaving strangely.
  2. ______
  3. ______

## Context

I am loading a script file dynamically which is getting loaded twice due to this.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office Online]: Outlook Web Client on Windows
* Host [Excel, Word, PowerPoint, etc.]: Outlook 365
* Office version number: Outlook 365
* Operating System: Windows
* Browser (if using Office Online): Chrome

## Useful logs

Nothing to provide here
- [ ] Console errors
- [ ] Screenshots
- [ ] Test file (if only happens on a particular file)

**Comment by exextoc** [comment]
Is this issue happening in old OWA or the new OWA client(After enabling toggle button)?

**Comment by kakusharma** [comment]
In old OWA, I guess. There is no toggle option there.

**Comment by exextoc** [comment]
What is the version of OWA? You can find this information using our Diagnostics API:
`Office.context.mailbox.diagnostics.hostVersion`

https://docs.microsoft.com/en-us/office/dev/add-ins/reference/objectmodel/preview-requirement-set/office.context.mailbox.diagnostics#hostversion-string

**Comment by kakusharma** [comment]
Its showing "2019080502.05"

**Comment by exextoc** [comment]
Does this repro when using `Office.initialize` instead of `Office.OnReady`?

**Comment by kakusharma** [comment]
Nope, it works fine with Office.initialize.

**Comment by exextoc** [comment]
Can you share the exact console error you are getting, when you are using Office.OnReady, and are you able to reproduce the same error in Outlook Desktop client as well, or does this happen only in OWA?

**Comment by kakusharma** [comment]
There is no error as such just that OnReady is being called twice. I have checked it on OWA only. I am unable to load the Add-In in Outlook Desktop client.

**Comment by exextoc** [comment]
Can you please share your Manifest and the basic add-in code in order for us to test. Thanks.

**Comment by kakusharma** [comment]
Manifest file:
`<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<OfficeApp xmlns="http://schemas.microsoft.com/office/appforoffice/1.1" 
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
  xmlns:bt="http://schemas.microsoft.com/office/officeappbasictypes/1.0" 
  xmlns:mailappor="http://schemas.microsoft.com/office/mailappversionoverrides/1.0" xsi:type="MailApp">
  <Id>05c2e1c9-3e1d-406e-9a91-e9ac64854143</Id>
  <Version>1.0.0.0</Version>
  <ProviderName>Proze</ProviderName>
  <DefaultLocale>en-US</DefaultLocale>
  <DisplayName DefaultValue="Proze Add-In"/>
  <Description DefaultValue="Add-In for best inline view experience."/>
  <IconUrl DefaultValue="https://localhost:8000/assets/icon-32.png"/>
  <HighResolutionIconUrl DefaultValue="https://localhost:8000/assets/icon-80.png"/>
  <SupportUrl DefaultValue="https://www.contoso.com/help"/>
  <AppDomains>
    <AppDomain>contoso.com</AppDomain>
  </AppDomains>
  <Hosts>
    <Host Name="Mailbox" />
  </Hosts>
  <Requirements>
    <Sets>
      <Set Name="Mailbox" MinVersion="1.1" />
    </Sets>
  </Requirements>
  <FormSettings>
    <Form xsi:type="ItemRead">
      <DesktopSettings>
        <SourceLocation DefaultValue="https://localhost:8000/index.html"/>
        <RequestedHeight>250</RequestedHeight>
      </DesktopSettings>
    </Form>
  </FormSettings>
  <Permissions>ReadItem</Permissions>
  <Rule xsi:type="RuleCollection" Mode="Or">
    <Rule xsi:type="ItemIs" ItemType="Message" FormType="Read" />
  </Rule>
  <DisableEntityHighlighting>false</DisableEntityHighlighting>
  <VersionOverrides xmlns="http://schemas.microsoft.com/office/mailappversionoverrides" xsi:type="VersionOverridesV1_0">
    <VersionOverrides xmlns="http://schemas.microsoft.com/office/mailappversionoverrides/1.1" xsi:type="VersionOverridesV1_1">
      <Requirements>
        <bt:Sets DefaultMinVersion="1.3">
          <bt:Set Name="Mailbox" />
        </bt:Sets>
      </Requirements>
      <Hosts>
        <Host xsi:type="MailHost">
          <DesktopFormFactor>
            <ExtensionPoint xsi:type="DetectedEntity">
              <Label resid="AddIn.Title" />
              <SourceLocation resid="AddIn.WebApp" />
              <Rule xsi:type="RuleCollection" Mode="And">
                <Rule xsi:type="ItemIs" ItemType="Message" />
                <Rule xsi:type="ItemHasRegularExpressionMatch" PropertyName="BodyAsPlaintext" RegExName="BodyFilter" RegExValue="Click here to (Play|Poll|View)" Highlight="all" />
              </Rule>
            </ExtensionPoint>
          </DesktopFormFactor>
        </Host>
      </Hosts>
      <Resources>
        <bt:Urls>
          <bt:Url id="AddIn.WebApp" DefaultValue="https://localhost:8000/index.html"/>
        </bt:Urls>
        <bt:ShortStrings>
          <bt:String id="AddIn.Title" DefaultValue="Proze Add-In"/>
        </bt:ShortStrings>
      </Resources>
    </VersionOverrides>
  </VersionOverrides>
</OfficeApp>`

Add-In code:
`Office.onReady().then(info => {
        console.log(info);
    });`

**Comment by kakusharma** [comment]
Also, getRegExMatchesByName is returning null for Outlook 2013 desktop client, even though its working well in web client. Is 'getRegExMatchesByName' available for desktop clients or not?

**Comment by exextoc** [comment]
@kakusharma I'm currently trying to reproduce your scenario, can you please file a separate issue for the getRegExMatchesByName returning null for Outlook 2013.

**Comment by kakusharma** [comment]
Sure, for that I already created a new issue.

**Comment by exextoc** [comment]
@kakusharma I'm unable to repro it in both Compose and Read mode

**Comment by kakusharma** [comment]
I think I found it. It was because the javascript file where I am initializing the Office was getting loaded twice. I just checked the source and found out. Sorry for the trouble. Closing the issue. Appreciate your help!
